src/Integration.Vsix/SonarLintDaemon/SonarLintDaemon.cs

```diff
@@ -242,28 +242,42 @@ private void WritelnToPane(string msg)
             logger.WriteLine(msg);
         }
 
-        // Need to be able to stub this method out for testing 
-        public virtual /* for testing */ void Stop()
+        public void Stop()
         {
             if (!IsRunning)
             {
                 return;
                 // throw exception?
             }
 
+            this.SafeInternalStop();
```

Comment:
We need to be able to do cleanup internally in cases where the process failed to start correctly i.e. when `IsRunning` might be return false. I moved the internal cleanup into a separate method.